src/ui/panes/queue.rs

```diff
@@ -82,24 +83,59 @@ impl Pane for QueuePane {
         } = context;
         let queue_len = queue.len();
 
+        let title = self
+            .filter
+            .as_ref()
+            .map(|v| format!("[FILTER]: {v}{} ", if self.filter_input_mode { "█" } else { "" }));
+
+        let table_block = {
+            let mut b = Block::default().border_style(config.as_border_style().bold());
+            if config.theme.show_song_table_header {
+                b = b.borders(Borders::TOP);
+            }
+            if let Some(ref title) = title {
+                b = b.title(title.clone().blue());
+            }
+            b
+        };
+
         let header_height = u16::from(config.theme.show_song_table_header);
+
+        let [data_area, scrollbar_area] =
+            Layout::horizontal([Constraint::Percentage(100), Constraint::Length(1)]).areas(area);
+
         let [table_header_section, mut queue_section] =
-            *Layout::vertical([Constraint::Min(header_height), Constraint::Percentage(100)]).split(area)
-        else {
-            return Ok(());
+            Layout::vertical([Constraint::Length(header_height), Constraint::Min(0)])
+                .horizontal_margin(1)
+                .areas(data_area);
+
+        let constraints = if config.theme.show_song_table_header {
+            vec![Constraint::Length(header_height + 1), Constraint::Min(0)]
```

Comment:
You can do `&[Constraint::Length(header_height + 1), Constraint::Min(0)] as &[Constraint]` in both branches to coerce them to the less specific slice type and avoid using vec allocation.

Reply:
Thanks for the tip! Fixed